src/parameter.rs

```diff
@@ -13,6 +13,43 @@ pub(crate) struct Parameter<'src> {
   pub(crate) name: Name<'src>,
 }
 
+impl<'src> Parameter<'src> {
+  pub(crate) fn format_eval<'this, 'eval, 'run>(
```

Comment:
Can we do something like this instead?

```rust
use super::*;

/// A single function parameter
#[derive(PartialEq, Debug, Clone, Serialize)]
pub(crate) struct Parameter<'src, T = Expression<'src>> {
  /// An optional default expression
  pub(crate) default: Option<T>,
  /// Export parameter as environment variable
  pub(crate) export: bool,
  /// The kind of parameter
  pub(crate) kind: ParameterKind,
  /// The parameter name
  pub(crate) name: Name<'src>,
}

impl<'src, T: Display> ColorDisplay for Parameter<'src, T> {
  fn fmt(&self, f: &mut Formatter, color: Color) -> Result<(), fmt::Error> {
    if self.export {
      write!(f, "$")?;
    }
    if let Some(prefix) = self.kind.prefix() {
      write!(f, "{}", color.annotation().paint(prefix))?;
    }
    write!(f, "{}", color.parameter().paint(self.name.lexeme()))?;
    if let Some(ref default) = self.default {
      write!(f, "={}", color.string().paint(&default.to_string()))?;
    }
    Ok(())
  }
}
```

This gives Parameter a generic parameter T which is the type of the default. T defaults to `Expression<'src>`, so everything works as normal. You can create an evaluated parameter with `Parameter<String>`.

So when `--eval` is passed, convert parameters to `Parameter<String>` using eval before printing.

Reply:
If `Parameter` becomes generic then when we evaluate it we need to store it separately to the originals because the recipes expect `Parameter<Expression<'src>>`, right? That would add some complexity to simply printing the recipes.
What if we instead evaluate them in place into `Expression::String`? That way the type stays the same, and we can keep most of the formatting code that was already in place, I think.

Reply:
I agree on the flag, I want to avoid changing something like the CLI in a breaking manner. Although, it feels a bit weird to have both `evaluate` and `eval`, do you think we can differentiate them more?